### PR TITLE
655 conditional text input

### PR DESCRIPTION
**Pull request by clemens-msupply**
Part of #655

- Adds an Autocomplete control
- Adds a ConditionalSelect control
- Use these controls to select province and district

The Autocomplete component supports a "freeStyle" mode which allows users to select from a dropdown as well as entering some text. This made this component a bit more complicated than expected. The build in freeSolo option worked but the onChange callback wasn't called when changing focus, i.e. the form data wasn't updated after entering the text. Furthermore, you have to handle a `string|DisplayOption` type when the input changed since the freeSolo option is always a string...

I tried to quickly write a test for it but had some problem to drill down to the right components to do the actions. Any experience with it? otherwise I will try again later.

### CODE REVIEW

**Comment by clemens-msupply** [comment]
Also closes #599

File: client/packages/common/src/ui/forms/JsonForms/components/Text.tsx [hunk 1]
```diff
@@ -109,6 +109,28 @@ const UIComponent = (props: ControlProps) => {
...
+            instancePath: path,
+            message: customErrors,
+            schemaPath: path,
+            keyword: '',
+            params: {},
+          },
+        ])
+      );
+    }
+  }, [core, dispatch, customErrors]);
```
Comment (clemens-msupply) [inline client/packages/common/src/ui/forms/JsonForms/components/Text.tsx:132]: If there was a custom validation and it failed you where still able to save the form

**Comment by clemens-msupply** [comment]
I also sneaked in a fix to track if data isDirty (at least for the patient form). Previously, leaving a saved form would trigger an unnecessary warning...

**Comment by mark-prins** [comment]
I was going to make an issue for the `isDirty` annoyance! thanks 🙏

**Review by mark-prins — APPROVED** [review]
👍  LGTM! working nicely

**Pull request merged**